databases/week3/data2.sql

```diff
@@ -0,0 +1,141 @@
+-- Meal queries
+
+--1) Get all meals
+
+SELECT *
+FROM meal;
+
+--2) Add a new meal
+
+INSERT INTO meal_sharing.meal (title, description, location, time, max_reservations, price, created_date)
+VALUES ('Tortilla', 'Bread dipped in fried eggs', 'Spain', '2020-12-30 20:20:00', '6', '15', '2020-12-20');
+
+--3) Get a meal with any id
+
+SELECT *
+FROM meal
+WHERE id = 3;
+
+--4) Update a meal with any id, fx 1. Update any attribute fx the title or multiple attributes
+
+UPDATE meal
+SET max_reservations = 10, price = 17
+WHERE id = 3;
+
+--5) Delete a meal with any id, fx 1
+
+DELETE FROM meal
+WHERE price < 10;
+
+--Reservation queries
+--1) Get all reservations
+
+SELECT *
+FROM reservation;
+
+--2) Add a new reservation
+
+INSERT INTO meal_sharing.reservation (number_of_guests, meal_id, created_date, contact_phonenumber, contact_name, contact_email) 
+VALUES ('5', '5', '2020-12-17', '202152', 'Loreta', 'lori@hello.com');
+
+--3) Get a reservation with any id, fx 1
+
+SELECT * 
+FROM reservation
+WHERE id = 4;
+
+--4) Update a reservation with any id, fx 1. Update any attribute fx the title or multiple attributes
+
+UPDATE reservation
+SET contact_email = 'felix@yahoo.dk'
+WHERE id = 5;
+
+
+--5) Delete a reservation with any id, fx 1
+
+DELETE FROM reservation
+WHERE contact_name = 'Mindi';
+
+--Review queries
+
+--1) Get all reviews
+
+SELECT *
+FROM review;
+
+--2) Add a new review
+
+INSERT INTO meal_sharing.review (title, description, meal_id, stars, created_date)
+VALUES ('All great', 'food is perfect', '3', '5', '2020-11-30');
+
+--3) Get a review with any id, fx 1
+
+SELECT *
+FROM review
+WHERE id = 5;
+
+--4) Update a review with any id, fx 1. Update any attribute fx the title or multiple attributes
+
+UPDATE review
+SET stars = 4
+WHERE id = 6;
+
+--5) Delete a review with any id, fx 1
+
+DELETE FROM review
+WHERE id = 7;
+
+-- Additional queries
+
+--Get meals that has a price smaller than a specific price
+
+SELECT *
+FROM meal
+WHERE price <= 20;
+
+--Get meals that still has available reservations
+
+SELECT meal.title, reservation.created_date AS reservation_date
+FROM meal
+JOIN reservation ON meal.id = reservation.meal_id
+WHERE reservation.created_date > '2020-11-01';
+
```

Comment:
The right answer is;

```
SELECT 
    meal.title,
    reservation.created_date AS reservation_date,
    meal.max_reservations,
    SUM(reservation.number_of_guests)
FROM
    meal
        JOIN
    reservation ON meal.id = reservation.meal_id
WHERE
    reservation.created_date > '2020-11-01'
GROUP BY meal.id , reservation.created_date
HAVING SUM(reservation.number_of_guests) < meal.max_reservations
```

Reply:
But why do I need to do all of that if the required is only to get the meals with available reservations? 
What is wrong with my code? (just asking to understand)

Reply:
OK sorry, apparently I totally miss-understood the question, but now I get it. Thanks for your feedback :)